zk_toolbox/crates/config/src/general.rs

```diff
@@ -162,6 +183,12 @@ fn update_port_in_url(http_url: &mut String, port: u16) -> anyhow::Result<()> {
     Ok(())
 }
 
+fn update_port_in_host(host: &mut Host, port: u16) -> anyhow::Result<()> {
+    let ip = host.0.split(':').next().context("Failed to get IP")?;
+    host.0 = format!("{}:{}", ip, port);
```

Comment:
":" is a valid character of the IPv6 address, so splitting on ':' won't work. Use https://github.com/matter-labs/era-consensus/blob/a298f504ac7f5c89e9dbc201721a89b1eeaa7663/node/libs/concurrency/src/net/mod.rs#L33 or Url::parse (https://docs.rs/url/latest/url/struct.Url.html)

Reply:
probably Url::parse will be more appropriate